src/containerapp/azext_containerapp/_up_utils.py

```diff
@@ -935,6 +936,19 @@ def _get_env_and_group_from_log_analytics(
                 env_list = [e for e in env_list if format_location(e["location"]) == format_location(location)]
             if env_list:
                 # TODO check how many CA in env
+                # Get container app list in resource_group
+                containerapp_list = list_containerapp(cmd=cmd, resource_group_name=resource_group_name)
+                if containerapp_list and len(containerapp_list) > 0:
+                    # Get container app in resource_group with same name as app
+                    container_app = next((c for c in containerapp_list if c.get("name", "").lower() == app.name.lower()), None) if app.name else None
```

Comment:
I think infer existing containerapp should be add in `_get_app_env_and_group`

```
def _get_app_env_and_group(
    cmd, name, resource_group: "ResourceGroup", env: "ContainerAppEnvironment", location
):
    matched_apps = []
    if not resource_group.name and not resource_group.exists:
        matched_apps = [c for c in list_containerapp(cmd) if c["name"].lower() == name.lower()]
    if resource_group.name and resource_group.exists:
        matched_apps = [c for c in list_containerapp(cmd, resource_group_name=resource_group.name) if c["name"].lower() == name.lower()]
        
    if env.name:
        matched_apps = [c for c in matched_apps if parse_resource_id(c["properties"]["environmentId"])["name"].lower() == env.name.lower()]
    if location:
        matched_apps = [c for c in matched_apps if format_location(c["location"]) == format_location(location)]
    if len(matched_apps) == 1:
        resource_group.name = parse_resource_id(matched_apps[0]["id"])[
            "resource_group"
        ]
        env.set_name(matched_apps[0]["properties"]["environmentId"])
    elif len(matched_apps) > 1:
        raise ValidationError(
            f"There are multiple containerapps with name {name} on the subscription. "
            "Please specify which resource group your Containerapp is in."
        )
```

Reply:
Hi @snehapar9 
The _`get_app_env_and_group`  is a more suitable place to fix this bug. What do you think?

Reply:
Synced offline. @Greedygre this makes sense to me. Reverted the change I mage in the _get_env_and_group_from_log_analytics method because we are setting the correct env by extracting from the container app if exists now  in the _get_app_env_and_group method.